### PR TITLE
Add an optional attribute to handle updates during atlas downloads

### PR DESCRIPTION
**Pull request by yoda-vid**
## Description

**What is this PR**

Atlas download progress is currently handled within a Rich progress indicator, but external libraries do not have access to these updates. This PR adds an optional attribute to `BrainGlobeAtlas` to reference a handler that is updated during atlas downloads. This handler can be given as a function that takes the currently competed and total required bytes to download the given atlas.

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**

Provides access to track real-time atlas download updates by external apps. There may be better ways to provide this functionality...I'm all ears to alternatives.

**What does this PR do?**

Adds a hook called by `bg-atlasapi` during atlas downloads.

## References

I am using this handler to update a progress bar in MagellanMapper when downloading atlases through BrainGlobe (see https://github.com/sanderslab/magellanmapper/pull/75).

## How has this PR been tested?

- Tested downloading the "kim_mouse_50um" atlas in bg-atlasapi in an interactive Python 3.6 session, where the handler is ignored and download proceeds as normal
- Tested in [MagellanMapper PR 75](https://github.com/sanderslab/magellanmapper/pull/75), which passes a function that updates the progress bar during atlas downloads

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

Updated the docstrings with the changes. Also added a type hint to the handler to specify its expected parameters.

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by yoda-vid** [comment]
Attempted to fix formatting errors from the lint action by running the `black` package.

**Review by adamltyson — APPROVED** [review]

**Comment by adamltyson** [comment]
Looks good to me @yoda-vid. Just because everything else depends on this package, I'll wait for @vigji or @FedeClaudi to approve before merging.

**Comment by yoda-vid** [comment]
Sounds good, thanks @adamltyson! Open to any feedback to make this best fit with BrainGlobe's goals.

**Review by vigji — APPROVED** [review]
Sorry for taking so long! Good to go for me.

**Comment by adamltyson** [comment]
Merged, thanks @yoda-vid!

**Pull request merged**
